public/New-DbaDbUser.ps1

```diff
@@ -69,165 +74,246 @@ function New-DbaDbUser {
+        Creates a new sql user named user1 for the login login1 in the database DB1.
 
     .EXAMPLE
-        PS C:\> New-DbaDbUser -SqlInstance sqlserver2014 -Database DB1 -Login Login1 -Username user1 -DefaultSchema schema1
+        PS C:\> New-DbaDbUser -SqlInstance sqlserver2014 -Database DB1 -User user1 -Login Login1 -DefaultSchema schema1
 
-        Creates a new sql user named user1 mapped to Login1 in the specified database and specifies the default schema to be schema1.
+        Creates a new sql user named user1 for the login login1 in the database DB1 and specifies the default schema to be schema1.
 
     .EXAMPLE
-        PS C:\> New-DbaDbUser -SqlInstance sqlserver2014 -Database DB1 -Username "claudio@********.onmicrosoft.com" -ExternalProvider
+        PS C:\> New-DbaDbUser -SqlInstance sqlserver2014 -Database DB1 -User "claudio@********.onmicrosoft.com" -ExternalProvider
 
-        Creates a new sql user named 'claudio@********.onmicrosoft.com' mapped to Azure Active Directory (AAD) in the specified database.
+        Creates a new sql user named 'claudio@********.onmicrosoft.com' mapped to Azure Active Directory (AAD) in the database DB1.
 
     .EXAMPLE
-        PS C:\> New-DbaDbUser -SqlInstance sqlserver2014 -Database DB1 -Username user1 -Password (ConvertTo-SecureString -String "DBATools" -AsPlainText)
+        PS C:\> New-DbaDbUser -SqlInstance sqlserver2014 -Database DB1 -Username user1 -Password (ConvertTo-SecureString -String "DBATools" -AsPlainText -Force)
 
-        Creates a new cointained sql user named user1 in the database given database with the password specified.
+        Creates a new contained sql user named user1 in the database DB1 with the password specified.
     #>
+
+    ### This command has more comments than other commands, because it should act as an example for other commands.
+    ### These extra lines start with "###" and should help new contributores to understand why we code the way we do.
+
+    ### All commands that change objects must use SupportsShouldProcess to support -WhatIf.
+    ### All commands that add new objects (New-...) must use ConfirmImpact = "Medium".
+    ### All commands that change existing objects (Set-...) must use ConfirmImpact = "???".
+    ### All commands that drop existing objects (Remove-...) must use ConfirmImpact = "High".
+    ### For most of the commands, we try to not use parameter sets and try to check valid parameter combinations inside of the command to be able to give the user a "nice" feedback.
+    ### But this is an example of a command that uses parameter sets, which gives better help output.
     [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Medium")]
     param(
-        [parameter(Mandatory, Position = 1)]
+        ### All commands that need a connection to an instance use the following two parameters in the way we define here.
+        ### This supports parameter by position for the parameter SqlInstance and makes this a mandatory parameter.
```

Comment:
this is more a "PS works this way" rather than "dbatools works this way". While I'm totally on the side of helping contributors, we can't help them if they don't know how PS parameter work (or, if we want to do that, a major crash course on PS params is needed, and ..... it wont' fit in comments here)

Reply:
The open question in this block is: Do we want SqlInstance to be a positional parameter? This is not the case in all commands.

Reply:
For now, let's support SqlInstance as the first and positional parameter. Can be discussed later outside of this PR.